src/main/java/core/tasks/Deadline.java

```diff
@@ -0,0 +1,98 @@
+package core.tasks;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import core.exceptions.NoDeadlineMentionedException;
+import core.exceptions.NoDescriptionGivenException;
+import utilities.OutputFormatter;
+
+/**
+ * The Deadline class.
+ *
+ * @author s7manth
+ * @version 0.1
+ */
+public class Deadline extends Task {
+    protected LocalDate byDate;
+    protected LocalTime byTime;
+    protected String by;
+
+    /**
+     * Constructor for the Deadline class.
+     *
+     * @param description The description of the deadline.
+     * @param by The time by which the deadline is supposed to be achieved.
+     */
+    private Deadline(String description, String by) {
+        super(description);
+        String[] dateAndTime = by.split(" ");
+        this.byDate = LocalDate.parse(dateAndTime[0]);
+        if (dateAndTime.length > 1) {
+            this.byTime = LocalTime.parse(dateAndTime[1]);
+        }
+        this.by = by;
+    }
```

Comment:
Would a variable name such as "endDate" or "endTime" be more descriptive to readers than "by"?